pennylane/_qubit_device.py

```diff
@@ -264,6 +264,8 @@ def execute(self, circuit, **kwargs):
             self._samples = self.generate_samples()
 
         multiple_sampled_jobs = circuit.is_sampled and self._has_partitioned_shots()
+        ret_types = [m.return_type for m in circuit.measurements]
```

Comment:
Reverted applying this suggestion because `ret_types` is being used more than once and the first iteration consumes the generator.
```py
In [1]: a = (i for i in range(5))

In [2]: a
Out[2]: <generator object <genexpr> at 0x7fe5be5c0f20>

In [3]: [i for i in a]
Out[3]: [0, 1, 2, 3, 4]

In [4]: list(a)
Out[4]: []
```